app/src/main/kotlin/com/aliucord/manager/ui/viewmodel/InstallViewModel.kt

```diff
@@ -54,55 +60,96 @@ class InstallViewModel(
 
                 log += "Downloading libs APK... "
                 downloadManager.downloadSplit(supportedVersion, "config.$libArch")
                 log += "Done\n"
-            }
+            }.copyTo(externalCacheDir.resolve("patched").resolve("config.$libArch-${supportedVersion}.apk"), true)
 
-            val localeApk = externalCacheDir.resolve("config.en-${supportedVersion}.apk").also { file ->
+            val localeApkFile = externalCacheDir.resolve("config.en-${supportedVersion}.apk").also { file ->
                 if (file.exists()) return@also run { log += "Using cached locale APK\n" }
 
                 log += "Downloading locale APK... "
                 downloadManager.downloadSplit(supportedVersion, "config.en")
                 log += "Done\n"
-            }
+            }.copyTo(externalCacheDir.resolve("patched").resolve("config.en-${supportedVersion}.apk"), true)
 
-            val xxhdpiApk = externalCacheDir.resolve("config.xxhdpi-${supportedVersion}.apk").also { file ->
+            val xxhdpiApkFile = externalCacheDir.resolve("config.xxhdpi-${supportedVersion}.apk").also { file ->
                 if (file.exists()) return@also run { log += "Using cached drawables APK\n" }
 
                 log += "Downloading drawables APK... "
                 downloadManager.downloadSplit(supportedVersion, "config.xxhdpi")
                 log += "Done\n"
-            }
+            }.copyTo(externalCacheDir.resolve("patched").resolve("config.xxhdpi-${supportedVersion}.apk"), true)
+
+            val (hermesLibrary, cppRuntimeLibrary) = run {
+                val latestHermesRelease = githubRepository
+                    .getHermesReleases().reduce { current, new ->
+                        if (Instant.parse(current.createdAt).isBefore(Instant.parse(new.createdAt))) new else current
+                    }
+                val hermesFile = externalCacheDir.resolve("hermes-release-${latestHermesRelease.tagName}.aar")
+                val cppRuntimeFile = externalCacheDir.resolve("hermes-cppruntime-release-${latestHermesRelease.tagName}.aar")
+                if (
+                    hermesFile.exists()
+                    && cppRuntimeFile.exists()
+                ) arrayOf(hermesFile, cppRuntimeFile).also { log += "Using cached patched libraries\n" }
+                else {
+                    log += "Downloading patched libraries... "
+
+                    arrayOf(
+                        downloadManager.download(
+                            url = latestHermesRelease.assets.find { it.name == "hermes-release.aar" }!!.browserDownloadUrl,
+                            fileName = hermesFile.name
+                        ),
+                        downloadManager.download(
+                            url = latestHermesRelease.assets.find { it.name == "hermes-cppruntime-release.aar" }!!.browserDownloadUrl,
+                            fileName = cppRuntimeFile.name
```

Comment:
You could split up downloading libs so its more readable and specific from a logging standpoint